sound/soc/intel/boards/sof_sdw.c

```diff
@@ -583,6 +532,17 @@ static int create_codec_dai_name(struct device *dev,
 
 		codec[comp_index].dai_name =
 			codec_info_list[codec_index].dai_name;
+
+		codec_conf[*codec_conf_index].dlc = *codec;
```

Comment:
Should be ``codec_conf[*codec_conf_index].dlc = codec[comp_index];`` otherwise, the codec_conf.dlc will be the first codec.

Reply:
yes, fixed. thanks @bardliao !